### PR TITLE
core/chains/terra/terratxm: limit terra_msgs queries

### PR DESCRIPTION
**Pull request by jmank88**
https://app.shortcut.com/chainlinklabs/story/29943

### CODE REVIEW

File: core/chains/terra/terratxm/txm.go [hunk 1]
```diff
@@ -125,58 +127,57 @@ func (txm *Txm) run() {
...
+			if err != nil {
+				// Should be impossible given the check in Enqueue
+				txm.lggr.Criticalw("failed to unmarshal msg, skipping", "err", err, "msg", m)
+				continue
+			}
+			m.ExecuteContract = &ms
+			_, err = sdk.AccAddressFromBech32(ms.Sender)
+			if err != nil {
+				// Should never happen, we parse sender on Enqueue
+				txm.lggr.Errorw("unable to parse sender", "err", err, "sender", ms.Sender)
```
Comment (RyanRHall) [inline core/chains/terra/terratxm/txm.go:153]: maybe `Criticalw`, for consistency

**Review by RyanRHall — APPROVED** [review]

**Review by connorwstein — CHANGES_REQUESTED** [review]

File: core/chains/terra/terratxm/txm.go [hunk 2]
```diff
@@ -125,58 +127,57 @@ func (txm *Txm) run() {
...
-	}
-	if max := txm.cfg.MaxMsgsPerBatch(); int64(len(unstarted)) > max {
-		unstarted = unstarted[:max+1]
-	}
-	txm.lggr.Debugw("building a batch", "batch", unstarted)
-	var msgsByFrom = make(map[string]terra.Msgs)
-	for _, m := range unstarted {
-		var ms wasmtypes.MsgExecuteContract
-		err := ms.Unmarshal(m.Raw)
+	for {
```
[the remark this answers is not in the document]
Comment (jmank88) [reply core/chains/terra/terratxm/txm.go:130]: Yeah I overlooked the slice with MaxMsgsPerBatch initially :+1:

**Comment by jmank88** [comment]
Hmmmm I wonder if that smoke test failure is 'real' from expecting messages faster than we send now...

**Comment by jmank88** [comment]
> Hmmmm I wonder if that smoke test failure is 'real' from expecting messages faster than we send now...

nvm, pre-existing

**Pull request merged**